example/lib/main.dart

```diff
@@ -42,77 +44,83 @@ void main() {
-    ],
-    theme: ThemeData(
-        useMaterial3: true,
-        bottomSheetTheme: const BottomSheetThemeData(
-          surfaceTintColor: Colors.white,
-          backgroundColor: Colors.white,
-        )),
-    routes: {
-      '/': (context) => const MyApp(),
-      '/dropInScreen': (context) => DropInScreen(
-            repository: AdyenDropInRepository(service: service),
-          ),
-      '/cardComponentScreen': (context) => const CardNavigationScreen(),
-      '/cardSessionComponentScreen': (context) => CardSessionComponentScreen(
-            repository: adyenCardComponentRepository,
-          ),
-      '/cardAdvancedComponentScreen': (context) => CardAdvancedComponentScreen(
-            repository: adyenCardComponentRepository,
-          ),
-      '/cardBottomSheetScreen': (context) => CardBottomSheetScreen(
-            repository: adyenCardComponentRepository,
-          ),
-      '/googlePayNavigation': (context) => const GooglePayNavigationScreen(),
-      '/googlePaySessionComponent': (context) =>
-          GooglePaySessionsComponentScreen(
-            repository: adyenGooglePayComponentRepository,
-          ),
-      '/googlePayAdvancedComponent': (context) =>
-          GooglePayAdvancedComponentScreen(
-            repository: adyenGooglePayComponentRepository,
-          ),
-      '/applePayNavigation': (context) => const ApplePayNavigationScreen(),
-      '/applePaySessionComponent': (context) => ApplePaySessionComponentScreen(
-            repository: adyenApplePayComponentRepository,
-          ),
-      '/applePayAdvancedComponent': (context) =>
-          ApplePayAdvancedComponentScreen(
-            repository: adyenApplePayComponentRepository,
-          ),
-      '/instantComponentNavigation': (context) =>
-          const InstantNavigationScreen(),
-      '/instantSessionComponent': (context) => InstantSessionComponentScreen(
-          repository: adyenInstantComponentRepository),
-      '/instantAdvancedComponent': (context) => InstantAdvancedComponentScreen(
-          repository: adyenInstantComponentRepository),
-      '/multiComponentNavigationScreen': (context) =>
-          const MultiComponentNavigationScreen(),
-      '/multiComponentSessionScreen': (context) => MultiComponentSessionScreen(
-            cardRepository: adyenCardComponentRepository,
-            googlePayRepository: adyenGooglePayComponentRepository,
-            applePayRepository: adyenApplePayComponentRepository,
-          ),
-      '/multiComponentAdvancedScreen': (context) =>
-          MultiComponentAdvancedScreen(
-            cardRepository: adyenCardComponentRepository,
-            googlePayRepository: adyenGooglePayComponentRepository,
-            applePayRepository: adyenApplePayComponentRepository,
-          ),
-      '/clientSideEncryption': (context) =>
-          CseScreen(repository: adyenCseRepository),
-    },
-    initialRoute: "/",
+  runApp(Provider(
+    notifier: CardStateNotifier(adyenCseRepository),
```

Comment:
Maybe move provide to CSE context to prevent confusion if merchant would copy-past code from example app.